artifactory/commands/transferfiles/utils.go

```diff
@@ -121,35 +159,67 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 			progressBar.SetRunningThreads(curProcessedUploadChunks)
 		}
 
-		// Each uploading thread receive a token from the source via the uploadTokensChan, so this go routine can poll on its status.
-		fillTokensBatch(awaitingStatusChunksSet, uploadTokensChan)
-		// When awaitingStatusChunksSet size is zero, it means that all the tokens are uploaded,
+		// Each uploading thread receive a token and a node id from the source via the uploadChunkChan, so this go routine can poll on its status.
+		fillChunkDataBatch(&chunksResilientManager, uploadChunkChan)
+		// When numChunksInMap size is zero, it means that all the tokens are uploaded,
 		// we received 'DONE' for all of them, and we notified the source that they can be deleted from the memory.
-		if awaitingStatusChunksSet.Size() == 0 {
+		// If during the polling some chunks data were lost due to network issues, either on the client or on the source,
+		// it will be written to the error channel
+		if chunksResilientManager.numChunksInMap == 0 {
 			if shouldStopPolling(doneChan) {
 				return
 			}
 			continue
 		}
 
-		chunksStatus, err := sendSyncChunksRequest(curTokensBatch, awaitingStatusChunksSet, deletedChunksSet, srcUpService)
+		chunksStatus, err := sendSyncChunksRequest(curTokensBatch, &chunksResilientManager, srcUpService)
 		if err != nil {
 			continue
 		}
 		// Clear body for the next request
 		curTokensBatch = UploadChunksStatusBody{}
-		removeDeletedChunksFromSet(chunksStatus.DeletedChunks, deletedChunksSet)
-		toStop := handleChunksStatuses(phaseBase, chunksStatus.ChunksStatus, progressBar, awaitingStatusChunksSet, deletedChunksSet, timeEstMng, errorsChannelMng)
+		removeDeletedChunksFromSet(chunksStatus.DeletedChunks, chunksResilientManager.deletedChunksSet)
+		toStop := handleChunksStatuses(phaseBase, &chunksStatus, progressBar, &chunksResilientManager, timeEstMng, errorsChannelMng)
 		if toStop {
 			return
 		}
 	}
 }
 
+// Compare between the number of chunks received from the latest syncChunks request to the chunks data we handle locally in nodeToChunksMap
+// If the number of the in progress chunks of a node within nodeToChunksMap differs from the chunkStatus received, There is missing data on the source side.
+// ErrorsChannelMg will receive this data, and it will be uploaded again in phase 3 or in an additional transfer file run.
+func checkChunkStatusSync(chunkStatus *UploadChunksStatusResponse, manager *ChunksResilientManager, errorsChannelMng *ErrorsChannelMng) {
```

Comment:
* I couldn't quite understand the logic inside this function. I suggest we meet to discuss it.
* After we agree that the logic is sound, let's also add a unit test for it.
* I think that the description of this function should be improved, because it it include some grammar errors.